.azure-pipelines/debug_builds.yml

```diff
@@ -0,0 +1,54 @@
+parameters:
```

Comment:
Is this file still needed?

Reply:
If we still want to set up a CI pipeline for windows-debug builds, yes. If we've decided that we don't want a pipeline for this, then we can remove this.

Reply:
What would be the use of the pipeline?

Reply:
To spot the kinds of issues fixed in this PR soon after they get merged, rather than waiting for a windows developer to hit them when trying to build in debug mode.

Reply:
If we wanted to do that, couldn't we just us a parameter on one of the existing pipelines, rather than maintaining a separate yml that will eventually break?

Reply:
I think we decided we didn't want to do that as it would then build on every commit, using up more azure resources than we wanted. This is set to build nightly only. If we've changed our mind on this I'm happy to adjust it to use a parameter on existing pipelines.

Reply:
I am not suggesting to adjust any parameters, rather, if we want to make a debug build, use a parameter of the existing pipelines.

Reply:
As far as I can tell from the azure docs it's not possible to use two different scheduling settings (i.e. cron and pr triggers) to schedule builds with different parameters in a single `.yml` file. My understanding from the docs is that they want you to use 2 separate `.yml` files for this (which might share a template). Since you were concerned this morning about using two `.yml` files I have now changed it to use parameters in the existing `.yml`, but this does mean our windows-debug builds will be triggered at the same time as all our other builds (i.e. for commits in every PR)

Also depends on the pipelines changes at https://github.com/scipp/pipelines/pull/22

Reply:
> but this does mean our windows-debug builds will be triggered at the same time as all our other builds (i.e. for commits in every PR)

As discussed previously, we cannot do that, since we are running out of builders.

> As far as I can tell from the azure docs it's not possible to use two different scheduling setting

Use the package build yml and also make debug builds of each package? That triggers only nightly, and sooner or later we may want to publish debug builds anyway (on a different conda label). Unless the debug build is published there is very limited use to it anyway, I suppose?